app/main.py

```diff
@@ -1,2 +1,118 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    next_node: "Node" = None
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+
+    def _hash(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._hash(key)
+        if self.hash_table[index] is None:
+            self.hash_table[index] = Node(key, value)
+        else:
+            current = self.hash_table[index]
+            while current:
+                if current.key == key:
+                    current.value = value
+                    return
+                if current.next_node is None:
+                    break
+                current = current.next_node
+            current.next_node = Node(key, value)
+        self.size += 1
+        if self.size / self.capacity > 0.7:
+            self._resize()
+
+    def _resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_hash_table = [None] * new_capacity
+        for i in range(self.capacity):
+            current = self.hash_table[i]
+            while current:
+                new_index = hash(current.key) % new_capacity
+                if new_hash_table[new_index] is None:
+                    new_hash_table[new_index] = Node(current.key,
+                                                     current.value)
```

Comment:
Look , your logic is repeating in the different methods ( same while loop , almost same cheks etc ) . Try to create some func to reuse your logic